Engine/Results/SymbolCapacity.cs

```diff
@@ -0,0 +1,279 @@
+                        _algorithm.UtcTime.Date == _previousOrderEvent.UtcTime.RoundUp(_resolution.ToTimeSpan());
+
+                default:
+                    timeout = _previousOrderEvent.UtcTime.AddHours(1);
+                    break;
+            }
+
+            return _algorithm.UtcTime <= timeout;
+        }
+
+        /// <summary>
+        /// Updates the market capacity of the Symbol. Called on each time step of the algorithm
+        /// </summary>
+        /// <returns>False if we're currently within the timeout period, True if the Symbol has went past the timeout</returns>
+        public bool UpdateMarketCapacity()
+        {
+            var bar = GetBar();
+            if (bar == null || bar.Volume == 0)
+            {
+                return false;
+            }
+
+            _previousBar = bar;
+
+            var utcTime = _algorithm.UtcTime;
+            var conversionRate = Security.QuoteCurrency.ConversionRate;
+            var timeBetweenBars = (decimal)(utcTime - (_previousTime ?? utcTime)).TotalMinutes;
+
+            if (_previousTime == null || timeBetweenBars == 0)
+            {
+                _previousTime = utcTime;
+                _previousVolume = bar.Volume;
+                _averageDollarVolume = conversionRate * bar.Close * bar.Volume;
+            }
+            else
+            {
+                _averageDollarVolume = ((bar.Close * conversionRate) * (bar.Volume + _previousVolume)) / timeBetweenBars;
+            }
+
+            _previousTime = utcTime;
+            _previousVolume = bar.Volume;
+
+            var includeMarketVolume = IncludeMarketVolume();
+            if (includeMarketVolume)
+            {
+                _marketCapacityDollarVolume += bar.Close * _fastTradingVolumeDiscountFactor * bar.Volume * conversionRate * Security.SymbolProperties.ContractMultiplier;
+            }
+
+            // When we've finished including market volume, signal completed
+            return !includeMarketVolume;
+        }
+
+        /// <summary>
+        /// Gets the TradeBar for the given time step. For Quotes, we convert
+        /// it into a TradeBar using market depth as a proxy for volume.
+        /// </summary>
+        /// <returns>TradeBar</returns>
+        private TradeBar GetBar()
+        {
+            TradeBar bar;
+            if (_algorithm.CurrentSlice.Bars.TryGetValue(_symbol, out bar))
+            {
+                return bar;
+            }
+
+            QuoteBar quote;
+            if (_algorithm.CurrentSlice.QuoteBars.TryGetValue(_symbol, out quote))
+            {
+                // Fake a tradebar for quote data using market depth as a proxy for volume
+                var volume = (quote.LastBidSize + quote.LastAskSize) / 2;
+                volume = _symbol.SecurityType == SecurityType.Forex
+                    ? _forexMinuteVolume
+                    : volume;
+
+                return new TradeBar(
```

Comment:
Feels like we need some consolidator + a rolling window

Reply:
Nevermind me didn't realize we were doing this only around an order event 👍

Reply:
Elected for a simpler 2 bar comparison since it was a fake bar, and only needed a couple of data points to estimate the average volume for the period.